Corona-Warn-App/src/main/java/de/rki/coronawarnapp/tracing/states/TracingState.kt

```diff
@@ -108,7 +108,7 @@ data class LowRisk(
         context.getString(R.string.risk_card_body_not_yet_fetched)
     }
 
-    fun getRiskContactBody(context: Context): String = if (daysWithEncounters == 0) {
+    fun getRiskContactBody(context: Context): String = if (daysWithEncounters == 0 || lastEncounterAt == null) {
```

Comment:
Not a fan of this solution. Fixes the symptom, not the cause.
We should look deeper at the data sources.
Looking at the log provided in the internal ticket, the last returned result is:

```kotlin
LastCombinedRiskResults(
	lastCalculated=CombinedEwPtRiskLevelResult(
		ptRiskLevelResult=PtRiskLevelResult(
			calculatedAt=2021-07-08T07:25:07.341Z,
			riskState=LOW_RISK, 
			presenceTracingDayRisk=[], 
			checkInWarningOverlaps=[]
		), 
		ewRiskLevelResult=RiskLevelTaskResult(
			calculatedAt=2021-07-08T07:25:08.026Z, 
			failureReason=null, 
			ewAggregatedRiskResult=EwAggregatedRiskResult(
				totalRiskLevel=LOW, 
				totalMinimumDistinctEncountersWithLowRisk=0, 
				totalMinimumDistinctEncountersWithHighRisk=0, 
				mostRecentDateWithLowRisk=null, 
				mostRecentDateWithHighRisk=null, 
				numberOfDaysWithLowRisk=0, 
				numberOfDaysWithHighRisk=0, 
				exposureWindowDayRisks=null
			), 
			exposureWindows.size=null
		), 
		exposureWindowDayRisks=[
			ExposureWindowDayRisk(dateMillisSinceEpoch=1625616000000, riskLevel=LOW, minimumDistinctEncountersWithLowRisk=2, minimumDistinctEncountersWithHighRisk=0), 
			ExposureWindowDayRisk(dateMillisSinceEpoch=1625529600000, riskLevel=HIGH, minimumDistinctEncountersWithLowRisk=2, minimumDistinctEncountersWithHighRisk=0)
		]
	), 
	lastSuccessfullyCalculated=CombinedEwPtRiskLevelResult(
		ptRiskLevelResult=PtRiskLevelResult(
			calculatedAt=2021-07-08T07:25:07.341Z, 
			riskState=LOW_RISK, 
			presenceTracingDayRisk=[], 
			checkInWarningOverlaps=[]), 
			ewRiskLevelResult=RiskLevelTaskResult(
			calculatedAt=2021-07-08T07:25:08.026Z, 
			failureReason=null, 
		ewAggregatedRiskResult=EwAggregatedRiskResult(
				totalRiskLevel=LOW, 
				totalMinimumDistinctEncountersWithLowRisk=0, 
				totalMinimumDistinctEncountersWithHighRisk=0, 
				mostRecentDateWithLowRisk=null, 
				mostRecentDateWithHighRisk=null, 
				numberOfDaysWithLowRisk=0, 
				numberOfDaysWithHighRisk=0, 
				exposureWindowDayRisks=null
			),
			exposureWindows.size=null
		),
		exposureWindowDayRisks=null
	)
)
```

`lastCalculated` having 
```kotlin
exposureWindowDayRisks=[
			ExposureWindowDayRisk(dateMillisSinceEpoch=1625616000000, riskLevel=LOW, minimumDistinctEncountersWithLowRisk=2, minimumDistinctEncountersWithHighRisk=0), 
			ExposureWindowDayRisk(dateMillisSinceEpoch=1625529600000, riskLevel=HIGH, minimumDistinctEncountersWithLowRisk=2, minimumDistinctEncountersWithHighRisk=0)
]
```

is likely the cause for the displayed encounters.
It's curious that `lastCalculated` contains these EWs while `lastSuccessfullyCalculated` doesn't. :thinking:

Reply:
I totally agree, it fixes the symptom, not the cause but I thought it was worth it, being such a small change in the code, at least for now. Reproducing the issue is quite tricky and it's the first time I've seen this reported. Maybe we could dedicate more time reproducing and investigating this bug at a later time. We could merge it for the sake of improving the user experience with the promise to have another look at a later point or we could just close it. It's up to you.

Reply:
lastSuccessfullyCalculated is only there for the risk state (in case the lastCalculated failed and we want to show the risk state before), not for getting any details. Details are only available for the lastCalculated

Reply:
A change I thought about but never implemented is to just expose lastSuccessfullyCalculatedState, to prevent misinterpretation